spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestViews.java

```diff
@@ -1335,6 +1335,35 @@ public void showViews() throws NoSuchTableException {
             row("global_temp", "globalviewforlisting", true), tempView);
   }
 
+  @Test
+  public void showViewsWithCurrentNamespace() {
+    String namespaceOne = "show_views_ns1";
+    String namespaceTwo = "show_views_ns2";
+    String viewOne = viewName("viewOne");
+    String viewTwo = viewName("viewTwo");
+    sql("CREATE NAMESPACE IF NOT EXISTS %s", namespaceOne);
+    sql("CREATE NAMESPACE IF NOT EXISTS %s", namespaceTwo);
+
+    // create one view in each namespace
+    sql("CREATE VIEW %s.%s AS SELECT * FROM %s.%s", namespaceOne, viewOne, NAMESPACE, tableName);
+    sql("CREATE VIEW %s.%s AS SELECT * FROM %s.%s", namespaceTwo, viewTwo, NAMESPACE, tableName);
+
+    Object[] v1 = row(namespaceOne, viewOne, false);
+    Object[] v2 = row(namespaceTwo, viewTwo, false);
+
+    sql("USE %s", namespaceOne);
+    // should only contain v1 but not v2
+    assertThat(sql("SHOW VIEWS")).contains(v1).doesNotContain(v2);
+    assertThat(sql("SHOW VIEWS IN %s", catalogName)).contains(v1).doesNotContain(v2);
```

Comment:
Does `SHOW VIEWS IN catalog` result in using the current namespace but in the specified catalog? Or does it list the root namespace of that catalog? I think it should either list the root namespace or default namespace of the given catalog. This seems incorrect to me.

Reply:
The behavior for `SHOW TABLES IN catalog_name` is to list the root namespace. If the root namespace is not a supported namespace, it fails (I tested a local HadoopCatalog). I think that is reasonable behavior: the `IN` clause specifies the namespace to list. The multi-part identifier passed to the `IN` clause is resolved and the current catalog is used if no catalog is specified, but the current namespace is not used.

Reply:
indeed this behavior results in listing the root namespace, which I missed. I've updated the code and the test accordingly to reflect that